connexion/api.py

```diff
@@ -84,12 +84,17 @@ def __init__(self, swagger_yaml_path, base_url=None, arguments=None,
                             'swagger_url': swagger_url,
                             'auth_all_paths': auth_all_paths})
         arguments = arguments or {}
-        with swagger_yaml_path.open() as swagger_yaml:
-            swagger_template = swagger_yaml.read()
+        with swagger_yaml_path.open(mode='rb') as swagger_yaml:
+            contents = swagger_yaml.read()
+            try:
+                swagger_template = contents.decode()
+            except UnicodeDecodeError:
+                swagger_template = contents.decode('utf-8', 'replace')
```

Comment:
No, because the keys need to be valid by Swagger/OpenAPI spec anyway. Please check the test with a invalid character. The yaml.load does not fail.